bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -58,51 +61,57 @@ def optimizedparam(sampleval, nfold=2):
 if args.make_signal_kde and args.make_template_kde:
     raise Exception("Choose only one option out of --make-signal-kde and --make-template-kde")
 
-#calling template data and constructing template_kde_hdf file over template params
-template_file = h5py.File(args.template_file, 'r')
-mass1_n = template_file['mass1'][:]
-mass2_n = template_file['mass2'][:]
-spin1z_n = template_file['spin1z'][:]
-spin2z_n = template_file['spin2z'][:]
-chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+#calling template data and constructing template_kde_hdf_file over same template params
+temp_file = h5py.File(args.template_file, 'r')
+approximant = temp_file['approximant'][:]
```

Comment:
Rather than assuming the template bank file always has 'approximant' and 'f_lower' datasets and no others except the masses and spins, it would be better to get a list of all the datasets in the bank file (whatever they are) and then loop over them getting their values from the file. Then the same datasets can be written into the output file.

Reply:
Also, since these datasets from the original bank file are always going to be present in the output file, both in the signal KDE and template KDE cases, it would be preferable to open the output file and write the datasets into it before doing any other computation.